In [2]:
# Loanding Data
#--------------
import pandas as pd
df = pd.read_csv('../data/cleaned_most_used_beauty_cosmetics_products_extended.csv')
df.head()

Product_Name              Brand        Category Usage_Frequency  \
0   Ultra Face Mask     Drunk Elephant           Blush          Weekly   
1    Ultra Lipstick      Laura Mercier  Makeup Remover      Occasional   
2       Ultra Serum     Natasha Denona     Highlighter           Daily   
3      Divine Serum        Ilia Beauty       Face Mask      Occasional   
4  Super Foundation  Charlotte Tilbury     Highlighter      Occasional   

   Price_USD  Rating  Number_of_Reviews Product_Size  Skin_Type Gender_Target  \
0      67.85     1.4                686         30ml  Sensitive        Female   
1     116.43     4.2               5483        250ml        Dry        Unisex   
2      90.84     1.6               5039        100ml  Sensitive          Male   
3      55.17     3.2               6202        250ml     Normal          Male   
4     140.56     1.7                297        100ml       Oily        Female   

   ... Gender_Target_Male Gender_Target_Unisex  Main_Ingredient_Aloe Vera  \
0  ...              False                False                      False   
1  ...              False                 True                      False   
2  ...               True                False                       True   
3  ...               True                False                      False   
4  ...              False                False                      False   

  Main_Ingredient_Glycerin  Main_Ingredient_Hyaluronic Acid  \
0                    False                            False   
1                    False                            False   
2                    False                            False   
3                     True                            False   
4                     True                            False   

   Main_Ingredient_Retinol  Main_Ingredient_Salicylic Acid  \
0                     True                           False   
1                    False                           False   
2                    False                           False   
3                    False                           False   
4                    False                           False   

   Main_Ingredient_Shea Butter  Main_Ingredient_Vitamin C  Product_Size_NoML  
0                        False                      False               30.0  
1                         True                      False              250.0  
2                        False                      False              100.0  
3                        False                      False              250.0  
4                        False                      False              100.0  

[5 rows x 28 columns]

## Análisis Univariante No Gráfico


**Objetivo:** Realizar analisis segun el tipo de variables; categoricas y numericas, interpretando la informacion de los datos a traves de las siguientes preguntas:

- *Variables categorícas*: Indentificar las mas y menos comunes
  1. ¿Qué marcas (Brand) tienen más productos?
  2. ¿Cuáles son las categorías (Category) más representadas? ¿Hay categorías que aparezcan muy poco?
  3. ¿Cuál es el uso más frecuente (Usage_Frequency) de los productos?
  4. ¿Qué tipo de piel (Skin_Type) es más objetivo de los productos?
  5. ¿Qué pais (Country_of_Origing) produce la mayor cantidad de productos ?
 
- *Variables que han sido encoded*:
  1. ¿Qué ingredientes (Main_Ingredient) son los más comunes?
  2. ¿Los productos están más orientados hacia un género específico?



In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15000 entries, 0 to 14999
Data columns (total 28 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   Product_Name                     15000 non-null  object 
 1   Brand                            15000 non-null  object 
 2   Category                         15000 non-null  object 
 3   Usage_Frequency                  15000 non-null  object 
 4   Price_USD                        15000 non-null  float64
 5   Rating                           15000 non-null  float64
 6   Number_of_Reviews                15000 non-null  int64  
 7   Product_Size                     15000 non-null  object 
 8   Skin_Type                        15000 non-null  object 
 9   Gender_Target                    15000 non-null  object 
 10  Packaging_Type                   15000 non-null  object 
 11  Main_Ingredient                  15000 non-null  object 
 12  Cruelty_Free      

In [4]:
# Divide the columns by type
variables_numericas = df.select_dtypes(include=[ 'float64', 'int64']).columns
variables_categoricas = df.select_dtypes(include=['object']).columns


In [5]:
df[variables_numericas]

Price_USD  Rating  Number_of_Reviews  Usage_Frequency_encoded  \
0          67.85     1.4                686                        2   
1         116.43     4.2               5483                        1   
2          90.84     1.6               5039                        3   
3          55.17     3.2               6202                        1   
4         140.56     1.7                297                        1   
...          ...     ...                ...                      ...   
14995     131.64     2.8               1470                        3   
14996      76.35     2.1               7652                        2   
14997      65.37     4.2                811                        1   
14998      59.93     4.8               5482                        2   
14999      36.96     1.1               5452                        3   

       Category_encoded  Packaging_Type_encoded  Product_Size_NoML  
0                    13                       5               30.0  
1                     0                       0              250.0  
2                    14                       1              100.0  
3                     3                       5              250.0  
4                    14                       1              100.0  
...                 ...                     ...                ...  
14995                 3                       5              250.0  
14996                 5                       2              250.0  
14997                14                       2               30.0  
14998                22                       2              250.0  
14999                13                       0              200.0  

[15000 rows x 7 columns]

In [6]:
df[['Price_USD', 'Rating',  'Number_of_Reviews',  'Product_Size_NoML']].describe()


Price_USD        Rating  Number_of_Reviews  Product_Size_NoML
count  15000.000000  15000.000000       15000.000000       15000.000000
mean      80.134108      3.002327        5014.231333         129.233333
std       40.402983      1.168029        2855.665464          78.544405
min       10.000000      1.000000          52.000000          30.000000
25%       45.480000      2.000000        2562.000000          50.000000
50%       80.040000      3.000000        5002.000000         100.000000
75%      114.760000      4.000000        7497.000000         200.000000
max      149.990000      5.000000       10000.000000         250.000000

#### ¿Qué marcas (Brand) tienen más productos?


In [7]:
brandsXproducts =  df.groupby('Brand').size().sort_values(ascending=False)
print(brandsXproducts)

Brand
Milk Makeup                426
Make Up For Ever           414
Kiehl’s                    411
NARS                       400
E.l.f.                     399
Farsali                    399
Yves Saint Laurent         394
Bite Beauty                392
Sisley                     392
Morphe                     392
KVD Beauty                 391
Glossier                   388
Becca                      388
Huda Beauty                384
RMS Beauty                 382
Bobby Brown                380
Ilia Beauty                379
Clinique                   377
Rare Beauty                375
Tatcha                     374
Too Faced                  371
Shiseido                   370
Kylie Cosmetics            370
Anastasia Beverly Hills    369
Drunk Elephant             368
Juvia’s Place              368
Natasha Denona             366
Fenty Beauty               366
Hourglass                  366
Charlotte Tilbury          364
Tarte                      361
Danessa Myricks            358
Bo

Respuesta: La marca con mas productos en el dataset es Milk Makeup, con un total de 426. Las que siguen son Make Up For Ever(414) y Kiehl’s(411) con una diferencia ligeramente menor.

 #### ¿Cuáles son las categorías (Category) más representadas? ¿Hay categorías que aparezcan muy poco?

In [8]:
df['Category'].value_counts().sort_values(ascending=False)

Category
Serum             710
Mascara           674
Face Oil          671
Face Mask         653
Highlighter       653
Moisturizer       638
Makeup Remover    636
Concealer         633
Lipstick          633
Bronzer           631
Eyeliner          629
Blush             628
Setting Spray     622
Cleanser          614
BB Cream          613
Foundation        613
Lip Liner         610
Primer            606
Contour           604
Powder            599
CC Cream          594
Exfoliator        583
Eye Shadow        582
Lip Gloss         571
Name: count, dtype: int64

La categoría con mayor cantidad de productos es el *Serum*, con un total de 710 productos, seguida por la masacara(674) y el face Oil (671).

El serum como top,podria estar relacionado a sus beneficios y presencias en el mercado impulsados en campañas publicitarias y redes sociales en los años anteriores al 2025. La máscara, por su parte, es un producto cuyo uso, es frecuente en la realizacion de maquillajes y su compra es de forma regular en el mercado.

Para el caso de Face oil, podria considerarse como una alternativa del serum , lo que nos lleva a preguntarnos: 
 ¿Su presencia en el top 3 es una respuesta a la demanda de los consumidores o es resultado de estrategias de marketing que promueven una amplia variedad de opciones?

La categoría con menor cantidad de productos es el *Lip Gloss*, con un total de 571 productos,podria considerarse que la diferencia con otras categorias no es muy grande, surge  sin embargo nos lleva a preguntarnos
 ¿Aunque es un producto usado en el maquillaje por que no se encuentra entre los primeros lugares es posible que se deba a la tendencias actuales en el maquillaje ?